### PR TITLE
Schemavalidation

### LINKED ISSUE #68: Add validation result field to the upload table, so faulty validation returns information to the user.

### PR DESCRIPTION
**Pull request by bmatern**
For task51, this adds the database column, REST endpoint, and webpage changes, for the HML validation feedback.

### CODE REVIEW

File: src/main/resources/config/liquibase/changelog/20200406120000_upload_validation_column.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.6.xsd
+                        http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+    <changeSet author="bmatern" id="addValidationColumn">
+        <addColumn
+                   tableName="upload">
+            <column name="validation_feedback" type="varchar(255)" defaultValue=""/>
+        </addColumn>
+    </changeSet>
+
```
Comment (cyt-mn) [inline src/main/resources/config/liquibase/changelog/20200406120000_upload_validation_column.xml:15]: Here you may want to add the new role to the jhi_authority table

File: src/main/java/org/ihiw/management/web/rest/UploadResource.java [hunk 2]
```diff
@@ -131,6 +131,52 @@ public UploadResource(UploadRepository uploadRepository, FileRepository fileRepo
...
+    /**
+     * {@code PUT  /uploads} : Set validation status on an existing upload.
+     *
+     * @param upload the upload to update.
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)} and with body the updated upload,
+     * or with status {@code 400 (Bad Request)} if the upload is not valid,
+     * or with status {@code 500 (Internal Server Error)} if the upload couldn't be updated.
+     * @throws URISyntaxException if the Location URI syntax is incorrect.
+     */
+    @PutMapping("/uploads/{id}")
```
Comment (cyt-mn) [inline src/main/java/org/ihiw/management/web/rest/UploadResource.java]: This should only be allowed for some sort of system user. Probably we want to create a new user (manually) with a new type of role "SYSTEM" (this has to go into AuthoritiesConstants.java). Following annotation should then do the job:

`@PreAuthorize("hasRole('" + AuthoritiesConstants.SYSTEM + "')")`

**Review by cyt-mn — CHANGES_REQUESTED** [review]
Security adjustments as described inline.

File: src/main/java/org/ihiw/management/web/rest/UploadResource.java (hunk 2 above)
Comment (bmatern) [reply src/main/java/org/ihiw/management/web/rest/UploadResource.java]: Done, the role is called VALIDATION

File: src/main/resources/config/liquibase/changelog/20200406120000_upload_validation_column.xml (hunk 1 above)
Comment (bmatern) [reply src/main/resources/config/liquibase/changelog/20200406120000_upload_validation_column.xml:15]: This has been done, ROLE_VALIDATION

**Review by bmatern — COMMENTED** [review]
Changes requested are done.

**Review by cyt-mn — APPROVED** [review]

**Pull request merged**